### PR TITLE
PCI-451 Implement create request handler that conforms to Open API spec

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Test and implement a controller method to handle a create certificate item request as per the Open API spec at https://developer-specs.kermit.aws.chdev.org/order-company-products/reference/certificates/create.
* No validation, no persistence at this stage.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,6 +54,12 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+		<dependency>
+			<groupId>org.hamcrest</groupId>
+			<artifactId>hamcrest-library</artifactId>
+			<version>2.1</version>
```
Comment (Mathsh) [inline pom.xml]: can you move the version to the properties like structured logging?

Comment (VaughanJackson123) [reply pom.xml]: Will do.

File: src/main/java/uk/gov/companieshouse/items/orders/api/model/ItemCosts.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return totalCost;
+    }
+
+    public void setTotalCost(String totalCost) {
+        this.totalCost = totalCost;
+    }
+
+    @Override
+    public String toString() {
+        return "ItemCosts{" +
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/items/orders/api/model/ItemCosts.java]: Can use Gson library for toString instead;

` @Override
    public String toString() {
        return new Gson().toJson(this);
    }`

File: src/main/java/uk/gov/companieshouse/items/orders/api/model/CertificateItemOptions.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    public boolean isCertShar() {
+        return isCertShar;
+    }
+
+    public void setCertShar(boolean certShar) {
+        isCertShar = certShar;
+    }
+
+    @Override
+    public String toString() {
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/items/orders/api/model/CertificateItemOptions.java]: Can use Gson library for toString instead;

` @Override
    public String toString() {
        return new Gson().toJson(this);
    }`

File: src/main/java/uk/gov/companieshouse/items/orders/api/dto/ItemDTO.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    @Override
+    public String toString() {
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/items/orders/api/dto/ItemDTO.java]: Can use Gson library for toString instead;

` @Override
    public String toString() {
        return new Gson().toJson(this);
    }`

File: src/main/java/uk/gov/companieshouse/items/orders/api/dto/CertificateItemDTO.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    public CertificateItemOptions getItemOptions() {
+        return itemOptions;
+    }
+
+    public void setItemOptions(CertificateItemOptions itemOptions) {
+        this.itemOptions = itemOptions;
+    }
+
+    @Override
+    public String toString() {
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/items/orders/api/dto/CertificateItemDTO.java]: Can use Gson library for toString instead;

` @Override
    public String toString() {
        return new Gson().toJson(this);
    }`

File: src/main/java/uk/gov/companieshouse/items/orders/api/model/ItemCosts.java (hunk 2 above)
Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/items/orders/api/model/ItemCosts.java]: Thanks, I'll try that.

**Pull request merged**